lib/auth.js

```diff
@@ -167,6 +167,17 @@ module.exports = class Auth {
                     }
                     return this.getAccessToken(forceRefresh, remainingAttempts);
                 } else {
+                    // Expired Error
+                    if (ex?.response?.data?.message) {
+                        ex.message = ex.response.data.message;
+                        ex.code = ex.response.data.errorcode;
+                    }
+                    // Unauthenticated
+                    else if (ex?.response?.data?.error_description) {
+                        ex.message = ex.response.data.error_description;
+                        ex.code = ex.response.data.error;
+                    }
```

Comment:
this could utilize the RESTError class that you could define in rest.js :)

Reply:
will avoid for now to avoid circular dependency. allows for additional checks in future also

Reply:
How about a dedicated custom error object like the util class in mcdev. And the import { ...} from statement in other classes